include/LightGBM/c_api.h

```diff
@@ -258,69 +306,83 @@ DllExport int LGBM_BoosterUpdateOneIter(BoosterHandle handle, int* is_finished);
 * \param grad gradient statistics
 * \param hess second order gradient statistics
 * \param is_finished 1 means finised(cannot split any more)
-* \return 0 when success, -1 when failure happens
+* \return 0 when succeed, -1 when failure happens
 */
 DllExport int LGBM_BoosterUpdateOneIterCustom(BoosterHandle handle,
   const float* grad,
   const float* hess,
   int* is_finished);
 
 /*!
-* \brief get evaluation for training data and validation data
+* \brief Rollback one iteration
 * \param handle handle
-* \param data 0:training data, 1: 1st valid data, 2:2nd valid data ...
-* \param out_len len of output result
-* \param out_result the string containing evaluation statistics, should allocate memory before call this function
-* \return 0 when success, -1 when failure happens
+* \return 0 when succeed, -1 when failure happens
 */
-DllExport int LGBM_BoosterEval(BoosterHandle handle,
-  int data,
-  int64_t* out_len,
-  float* out_results);
+DllExport int LGBM_BoosterRollbackOneIter(BoosterHandle handle);
+
+/*!
+* \brief Get iteration of current boosting rounds
+* \return iteration of boosting rounds
+*/
+DllExport int LGBM_BoosterGetCurrentIteration(BoosterHandle handle, int64_t* out_iteration);
 
 /*!
-* \brief get raw score for training data, used to calculate gradients outside
+* \brief Get number of eval 
+* \return total number of eval result
+*/
+DllExport int LGBM_BoosterGetEvalCounts(BoosterHandle handle, int64_t* out_len);
+
+/*!
+* \brief Get number of eval
+* \return total number of eval result
+*/
+DllExport int LGBM_BoosterGetEvalNames(BoosterHandle handle, int64_t* out_len, char** out_strs);
+
+/*!
+* \brief get evaluation for training data and validation data
 * \param handle handle
+* \param data_idx 0:training data, 1: 1st valid data, 2:2nd valid data ...
 * \param out_len len of output result
-* \param out_result used to set a pointer to array
-* \return 0 when success, -1 when failure happens
+* \param out_result the string containing evaluation statistics, should allocate memory before call this function
+* \return 0 when succeed, -1 when failure happens
 */
-DllExport int LGBM_BoosterGetScore(BoosterHandle handle,
+DllExport int LGBM_BoosterGetEval(BoosterHandle handle,
+  int data_idx,
   int64_t* out_len,
```

Comment:
I guess the parameter `out_len` is not really required anymore for this function, nor for any of the other functions that have an `out_results` parameter that needs to be allocated by the caller. As far as I can see, every function that has this signature, have an accompanying function that can be called to get the exact length of the array that should be allocated (e.g. `LGBM_BoosterGetEvalCounts()` here). So it would be rather wasteful to rely on `out_len` to shrink an over-allocated array after the call. Or are there any other cases in which this parameter would be useful?